plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -186,3 +193,63 @@ func (meta *predicateMetadata) ShallowCopy() algorithm.PredicateMetadata {
 		meta.serviceAffinityMatchingPodList...)
 	return (algorithm.PredicateMetadata)(newPredMeta)
 }
+
+func (pfactory *PredicateMetadataFactory) getMatchingAntiAffinityTerms(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) (map[string][]matchingPodAntiAffinityTerm, error) {
+	allNodeNames := make([]string, 0, len(nodeInfoMap))
+	for name := range nodeInfoMap {
+		allNodeNames = append(allNodeNames, name)
+	}
+
+	var lock sync.Mutex
+	var firstError error
+	result := make(map[string][]matchingPodAntiAffinityTerm)
+	appendResult := func(toAppend map[string][]matchingPodAntiAffinityTerm) {
+		lock.Lock()
+		defer lock.Unlock()
+		for uid, terms := range toAppend {
+			result[uid] = append(result[uid], terms...)
+		}
+	}
+	catchError := func(err error) {
+		lock.Lock()
+		defer lock.Unlock()
+		if firstError == nil {
+			firstError = err
+		}
+	}
+
+	processNode := func(i int) {
+		nodeInfo := nodeInfoMap[allNodeNames[i]]
+		node := nodeInfo.Node()
+		if node == nil {
+			catchError(fmt.Errorf("node not found"))
+			return
+		}
+		nodeResult := make(map[string][]matchingPodAntiAffinityTerm)
+		for _, existingPod := range nodeInfo.PodsWithAffinity() {
+			affinity := existingPod.Spec.Affinity
+			if affinity == nil {
+				continue
+			}
+			for _, term := range getPodAntiAffinityTerms(affinity.PodAntiAffinity) {
+				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pfactory.namespaceLister, existingPod, &term)
+				selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
+				if err != nil {
+					catchError(err)
+					return
+				}
+				if priorityutil.PodMatchesTermsNamespaceAndSelector(pod, namespaces, selector) {
```

Comment:
Instead of reading all namespaces that match the term.namespaceSelector and check whether the pod's namespace is among them, why don't we get pod's namespace and check whether pod's namespace matches term.namespaceSelector?

Reply:
@bsalamat Good point, I will take a look later.

Reply:
@bsalamat namespaceSelector can be used to match the lables of the pod namespace, so if we want to do this, we need all the api to get the pod namespace object and retrive the labels, then try the match operation, I didn't see any benefit here.

Reply:
@wanghaoran1988 I am not sure if I follow the concern. I understand that the flow of the code and the parameters that you pass to the PodMatchesTermNamespaceAndSelector will be different, but there is no major problem with doing that.  PodMatchesTermNamespaceAndSelector already receives "pod". It can get `term.NamespaceSelector` as well and check whether `pod` matches the selector. Am I missing something?

Reply:
I mean the `term.NamespaceSelector` are using select `namespace.Labels`, so we need the namespaceInformer to get the `Namespace` object using the `pod.Namespace`, so my way: get all namespaces and check whether the pod's namespace whether is among them and yours: get the pods namespace Labels and check whether it's matching the namespaceSelector,  I don't know which one is the best, but I don't understand your way's benifit.

Reply:
@wanghaoran1988 If I understand correctly, @bsalamat's concern is that iterating over all namespaces and checking the selector is not needed. As your earlier comment states, you would have to somehow fetch the pod's namespace object. It would take some extra code, but that is probably worth it to avoid iterating over all namespaces.

Reply:
@wanghaoran1988 I see your point. I guess the efficiency of the two approaches depends on the size of clusters and how many pods with such affinities exist. Your current approach is fine. We can profile and optimize it in the future if we run into a serious issue.